Source/WebCore/animation/ViewTimeline.cpp

```diff
@@ -74,13 +78,56 @@ Ref<ViewTimeline> ViewTimeline::createFromCSSValue(Style::BuilderState& builderS
     return adoptRef(*new ViewTimeline(nullAtom(), axis, { startInset, endInset }));
 }
 
-// FIXME: compute offset values from options.
+static std::optional<Length> lengthForInset(std::variant<RefPtr<CSSNumericValue>, RefPtr<CSSKeywordValue>> inset)
+{
+    // TODO: Figure out how to convert CSSNumericValue to Length
+    if (holds_alternative<RefPtr<CSSNumericValue>>(inset))
+        auto value = get<RefPtr<CSSNumericValue>>(inset);
+    else if (holds_alternative<RefPtr<CSSKeywordValue>>(inset))
+        auto value = get<RefPtr<CSSKeywordValue>>(inset);
+    return { };
+}
+
+static Length lengthForInsetCSSValue(RefPtr<CSSPrimitiveValue> value)
+{
+    if (!value)
+        return { };
+    if (value->isPercentage())
+        return Length(value->resolveAsPercentageNoConversionDataRequired(), LengthType::Percent);
+    if (value->isPx())
+        return Length(value->resolveAsLengthNoConversionDataRequired<int>(), LengthType::Fixed);
```

Comment:
Evidently this leads to a crash in `imported/w3c/web-platform-tests/scroll-animations/view-timelines/unattached-subject-inset.html` due to an `!isCalculated()` assertion. Not sure why we're hitting this since that test doesn't seem to use `calc()`.

Reply:
This test does actually use CSSMathNegate which might be causing this. I'm confused as to how we're hitting the string codepath for this provided inset though. Looking into this.

Reply:
Fun `new CSSMathNegate(CSS.px(144))` comes in as a `calc(-144px)` string. Seems like it should be ending up as  a `CSSNumericValue` but maybe this is expected behavior. For now I early return if we get a calc value since I think we need conversion data to be able to convert it to a length.

Reply:
Cc @weinig who's been doing work in that area recently.